### PR TITLE
Added safety checks to fishhook.c

### LINKED ISSUE #25182: [Xcode 11] [iOS 13] EXC_BAD_ACCESS in fishhook.c
# Bug

When running my app on an iPhone XR (iOS 13), the app immediately crashes with EXC_BAD_ACCESS. 

Screen Shot 2019-06-06 at 5 04 19 PM

I can only reproduce this when running on a physical iPhone. iPad and simulator both work fine.

## Environment info

React native info output:

```bash
 info
  React Native Environment Info:
    System:
      OS: macOS 10.15
      CPU: (4) x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
      Memory: 1.78 GB / 16.00 GB
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 8.16.0 - ~/.nvm/versions/node/v8.16.0/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v8.16.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 13.0, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 11.0/11M336w - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.6 => 16.8.6
      react-native: ^0.59.9 => 0.59.9
```

Library version: 0.59.9

## Steps To Reproduce

Make a new React Native project with `react-native init` and run on device.

Describe what you expected to happen:

1. App to not crash immediately

## Reproducible sample code

[Here](https://github.com/chrisspankroy/crash_test) That's a link to a repo I just made for a test project.

EDIT: Fixed the link to the repo

### PR DESCRIPTION
**Pull request by chrisspankroy**
## Summary

Fixes a crash that would occur on devices running iOS 13. `fishhook.c` would access bad memory, this adds safety checks to the offending line. This fixes #25182 
## Changelog

[iOS] [Fixed] - Add safety checks to `fishhook.c` to prevent bad memory access

## Test Plan

Build and run test suite

### CODE REVIEW

**Comment by chrisspankroy** [comment]
I don't think the failing tests are related to this but this is my first PR here so I could be missing something

File: Libraries/fishhook/fishhook.c [hunk 1]
```diff
@@ -103,7 +103,11 @@ static void perform_rebinding_with_section(struct rebindings_entry *rebindings,
               indirect_symbol_bindings[i] != cur->rebindings[j].replacement) {
             *(cur->rebindings[j].replaced) = indirect_symbol_bindings[i];
           }
-          indirect_symbol_bindings[i] = cur->rebindings[j].replacement;
+          if (i < ( sizeof(indirect_symbol_bindings) / sizeof(indirect_symbol_bindings[0]))) {
+            if (cur != NULL) {
```
Comment (sammy-SC) [inline Libraries/fishhook/fishhook.c]: why is this necessary if `cur` is being checked on line 99?

Comment (chrisspankroy) [reply Libraries/fishhook/fishhook.c]: Good point, I didn't see that. I'll fix it and test in a little bit

Comment (chrisspankroy) [reply Libraries/fishhook/fishhook.c]: Fixed

**Comment by hramos** [comment]
Should this change be made to https://github.com/facebook/fishhook upstream first?

**Comment by chrisspankroy** [comment]
I can open a PR there too 😃

**Comment by chrisspankroy** [comment]
Opened

File: Libraries/fishhook/fishhook.c [hunk 2]
```diff
@@ -103,7 +103,9 @@ static void perform_rebinding_with_section(struct rebindings_entry *rebindings,
               indirect_symbol_bindings[i] != cur->rebindings[j].replacement) {
             *(cur->rebindings[j].replaced) = indirect_symbol_bindings[i];
           }
-          indirect_symbol_bindings[i] = cur->rebindings[j].replacement;
+          if (i < ( sizeof(indirect_symbol_bindings) / sizeof(indirect_symbol_bindings[0]))) {
+              indirect_symbol_bindings[i] = cur->rebindings[j].replacement;
+          }
```
Comment (zhongwuzw) [inline Libraries/fishhook/fishhook.c]: Why `indirect_symbol_bindings` would out of bounds? If we bypass the assignment, hook would be invalid IMO.

**Comment by cpojer** [comment]
Thanks @chrisspankroy. @mmmulani tells me we can kill this code all together. I'll close this PR and you can expect a change to be committed to master soon that removes this code.

**Comment by chrisspankroy** [comment]
Works for me, thanks 👍

**Comment by mf168** [comment]
Works for me, thanks 👍
